src/main/java/uk/gov/hmcts/reform/sscs/service/PdfStoreService.java

```diff
@@ -49,10 +64,35 @@ public PdfStoreService(EvidenceManagementService evidenceManagementService) {
                     .build();
             SscsDocument pdfDocument = SscsDocument.builder().value(sscsDocumentDetails).build();
 
-            return Collections.singletonList(pdfDocument);
+            return singletonList(pdfDocument);
+        } catch (RestClientException e) {
+            log.error("Failed to store pdf document but carrying on [" + fileName + "]", e);
+            return emptyList();
+        }
+    }
+
+    public List<SscsDocument> storeSecureDocStore(byte[] content, String fileName, String documentType, SscsDocumentTranslationStatus documentTranslationStatus) {
+        ByteArrayMultipartFile file = ByteArrayMultipartFile.builder().content(content).name(fileName)
+                .contentType(APPLICATION_PDF).build();
+        try {
+            IdamTokens idamTokens = idamService.getIdamTokens();
```

Comment:
Here we always get the token of system.update@hmcts.net user. Don't we need to use the actual user that upload the document?

Reply:
For this case, the system is the user that uploads the email pdf to the appeal. I think it's only used in notifications.

You could argue the person that triggered the event also sends the email and should be the person the system should use to upload the notification to the appeal. Is that what you think we should do?

Reply:
Yes, I thought that's the requirement. May be I am wrong. If we use the system user for both uploading and downloading, then that should be fine.